File: solutions/4284-smallest-stable-index-i/smallest-stable-index-i.js
var firstStableIndex = function (nums, k) {
    const n = nums.length;
    for (let i = 0; i < n; i++) {
        let maxValue = nums[i];
        let minValue = nums[i];
        for (let j = 0; j < i; j++) {
            maxValue = Math.max(maxValue, nums[j]);
        }
        for (let j = i + 1; j < n; j++) {
            minValue = Math.min(minValue, nums[j]);
        }
        if (maxValue - minValue <= k) {
            return i;
        }
    }
    return -1;
};

File: solutions/4284-smallest-stable-index-i/smallest-stable-index-i.rs
impl Solution {
    pub fn first_stable_index(nums: Vec<i32>, k: i32) -> i32 {
        let n = nums.len();
        for i in 0..n {
            let mut max_value = nums[i];
            let mut min_value = nums[i];
            for j in 0..i {
                max_value = max_value.max(nums[j]);
            }
            for j in i + 1..n {
                min_value = min_value.min(nums[j]);
            }
            if max_value - min_value <= k {
                return i as i32;
            }
        }
        -1
    }
}